edit blessing steps

ssoonmi opened the pull request:
add edit blessing steps menu and page, change editGuest route to include personId in the path, allow adding, deleting, changing and undo for blessing steps

ssoonmi committed:
Add edit blessing steps menu and page

change editGuest route to include personId in the path
allow adding, deleting, changing and undo for blessing steps

ssoonmi committed:
Add spouse to action when adding a blessing action on person

anjosh-zz reviewed (changes requested):
The undo is slick I like it. Looks like that all works. Though I am a bit concerned about the value of the undo on this page since there is a Done button and there's a chance that someone might make a change and expect it to not save and navigate away and it does save. Not sure what we can do about it at the moment. I'll think about it.

A few comments:

- We have an edit button and edit blessing steps. that is a bit confusing. I'd like us to consolidate that. Also, we still have the edit single blessing step which is not necessary now.
- Could you move the undo button to left side since right side represents moving forward and left backward.
- Doesn't update blessing steps for the spouse.

anjosh commented:
Make edit blessing steps to not be all caps.
Make the menu icon the same size as the others.
Remove link to edit blessing step.

ssoonmi committed:
Make changes to Edit Blessing Steps feature

Remove edit button on each blessing step on person personProfile
Move undo button on edit blessing step page to left side
Make ellipsis menu icon on personProfile same size as others
Make edit blessing steps in personProfile menu to not be all caps.

ssoonmi commented:
To do:
- create a saved flag next to the Done button so people know that it is autosaving

ssoonmi committed:
Fix tooltip and click feature on personProfile

ssoonmi committed:
Add Saved text for EditBlessingSteps

ssoonmi commented:
Implemented saved flag

anjosh-zz reviewed (changes requested):
Thanks for making the changes I asked for. It looks like it works well. I have a few comments.

Would you mind going over the logic for how undo works for me? I'd really appreciate it because it would save me a lot of time! :) Talk to you tomorrow.


One cosmetic thing I noticed is:
<img width="1440" alt="Screen Shot 2019-11-21 at 4 03 05 PM" src="https://user-images.githubusercontent.com/2905269/69376548-76fb2200-0c78-11ea-8aff-b4be69f81c3d.png">
Looks like it goes to it's own row in certain cases.

anjosh-zz commented on pages/editBlessingSteps/_personId.vue:
```diff
@@ -0,0 +1,214 @@
+            spouse = spouseRelationship.Subject
+          }
+          if (spouse && spouse.fullname) {
+            this.person.spouse = spouse
```
Done

anjosh-zz commented on components/personProfile.vue:
```diff
@@ -42,7 +61,9 @@
-            <v-list-tile @click="item.handleClick">
+            <v-list-tile 
+              v-on:click="() => item.handleClick ? item.handleClick() : null"
+              :class="item.handleClick ? '' : 'unclickable'">
```
tried and didn't work on first try. not going to work on this
